widgets/textinput/options.go

```diff
@@ -269,6 +270,19 @@ func OnSubmit(fn SubmitFn) Option {
 	})
 }
 
+// ChangeFn if provided is called when the content of the text input field changes,
+// the argument data contains all the text in the field.
+//
+// The callback function must be thread-safe as the keyboard event that
+// triggers the submission comes from a separate goroutine.
+type ChangeFn func(data string)
+
+func OnChange(fn ChangeFn) Option {
```

Comment:
Let's document this option as well, since it is public and exported.

We could probably split the above comment into two. Leave the portion that describes the argument of the ChangeFn next to it and describe the option itself here.